### PR TITLE
React Highlight Board

### PR DESCRIPTION
**Pull request by bobwise**
Issue https://github.com/sparkdesignsystem/spark-design-system/issues/1073

## What does this PR do?
Add SprkHighlightBoard component for React.

SprkHighlightBoard does not have a subtitle or content property. The `children` property will be ignored by the React component.

Also removing remnants of the old "fullwidth" variant. Full width can be applied to the component by adding a Spark utility class.

### Documentation
 - [x] Update Spark Docs React
 - [x] Update Spark Docs Vanilla

### Code
 - [x] Build Component in Spark React
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)

### CODE REVIEW

File: src/patterns/components/highlight-board/react/code/default.hbs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<SprkHighlightBoard
+  imgSrc="https://staging.sparkdesignsystem.com/assets/toolkit/images/desktop.jpg"
+  imgAlt='flower'
+  heading='Hello, Welcome To Spark Design System'
+  ctaText='Button 1'
+  ctaHref='#'
+  ctaText2='Button 2'
+  ctaHref2='#'/>
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/code/default.hbs]: Hey @bobwise, these React Code examples need to match the content that's displayed in the site. So, when they are used - they resemble the rendering we have in the doc site. Button 1 and Button 2 / alt text, ETC ETC

File: src/patterns/components/highlight-board/react/code/no-image.hbs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<SprkHighlightBoard
+  variant='noimage'
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/code/no-image.hbs]: 'noimage' needs to have casing, I would suggest camelCase.

File: src/patterns/components/highlight-board/react/code/stacked.hbs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<SprkHighlightBoard
+  variant='stacked'
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/code/stacked.hbs]: Where's the prop for data-id?

File: src/patterns/components/highlight-board/react/info/default.hbs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">
+        heading
+      </td>
+      <td>
+        string
+      </td>
+      <td>
+        The heading text
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/info/default.hbs]: These need full stops at the end of these sentences.

File: src/react/projects/spark-core-react/src/SprkHighlightBoard/SprkHighlightBoard.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import SprkHighlightBoard from './SprkHighlightBoard';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+it('should display a div element with the correct base class', () => {
+  const wrapper = shallow(<SprkHighlightBoard />);
+  expect(wrapper.find('div.sprk-c-HighlightBoard').length).toBe(1);
+});
+
+it('should not render an image if imgSrc is not provided', () => {
```
Comment (afebbraro) [inline src/react/projects/spark-core-react/src/SprkHighlightBoard/SprkHighlightBoard.test.js]: I would like to make imgAlt required if imgSrc is provided. Think you could look into doing that?

Comment (bobwise) [reply src/react/projects/spark-core-react/src/SprkHighlightBoard/SprkHighlightBoard.test.js]: Custom validators coming in a later PR

File: src/angular/projects/spark-extras-angular-highlight-board/src/lib/spark-extras-angular-highlight-board.component.ts [hunk 6]
```diff
@@ -104,11 +104,6 @@ export class SparkHighlightBoardComponent {
           'sprk-c-HighlightBoard--has-image sprk-c-HighlightBoard--stacked'
         );
         break;
-      case 'fullWidth':
```
Comment (afebbraro) [inline src/angular/projects/spark-extras-angular-highlight-board/src/lib/spark-extras-angular-highlight-board.component.ts]: Hey @bobwise, please add some information to your PR about why there are Angular changes.

File: src/patterns/components/highlight-board/react/info/default.hbs [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">
+        children
+      </td>
+      <td>
+        Any renderable element(s)
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/info/default.hbs]:
```suggestion
        Any renderable element(s).
```

File: src/patterns/components/highlight-board/react/info/default.hbs [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">
+        ctaText
+      </td>
+      <td>
+        string
+      </td>
+      <td>
+        The text for the first CTA.
```
Comment (afebbraro) [inline src/patterns/components/highlight-board/react/info/default.hbs]: We should define CTA here.

File: src/patterns/components/highlight-board/react/info/default.hbs (hunk 7 above)
Comment (bobwise) [reply src/patterns/components/highlight-board/react/info/default.hbs]: removed this whole section - React component doesn't use the `children` property.

File: src/patterns/components/highlight-board/react/info/default.hbs (hunk 8 above)
Comment (bobwise) [reply src/patterns/components/highlight-board/react/info/default.hbs]: replaced with "button". These CTAs will always render as a button, never a link.

**Comment by afebbraro** [comment]
Good to merge! @bobwise

**Pull request merged**